tests/modules/submissions/resources/test_ensure_submission.py

```diff
@@ -53,3 +53,60 @@ def test_ensure_clone_submission_by_uuid(
         raise ex
     finally:
         clone.cleanup()
+
+
+def test_ensure_permissions(
+    flask_app_client, regular_user, readonly_user, db, test_clone_submission_uuid
+):
+    # Clone as the regular user and then try to reread as both regular and readonly user,
+    # read by regular user should succeed, read by readonly user should be blocked
+
+    clone = clone_submission(
+        flask_app_client, regular_user, test_clone_submission_uuid, later_usage=True
+    )
+
+    try:
+        with flask_app_client.login(
+            regular_user,
+            auth_scopes=(
+                'submissions:read',
+                'assets:read',
+            ),
+        ):
+            submission_response = flask_app_client.get(
+                '/api/v1/submissions/%s' % test_clone_submission_uuid
+            )
+            asset_response = flask_app_client.get(
+                '/api/v1/assets/3abc03a8-39c8-42c4-bedb-e08ccc485396'
```

Comment:
We may want to parameter the UUID in the future

Reply:
Agreed. I thought that as I copy & bodged it from the test above :)
Replace test_clone_submission_uuid with test_clone_submission that returns the UUID to be cloned and the UUIDs of the assets that it should contain, I guess. I'll have a look at that.